packages/fx-core/src/core/middleware/projectMigratorV3.ts

```diff
@@ -923,6 +923,20 @@ export async function generateApimPluginEnvContent(context: MigrationContext): P
   }
 }
 
+export async function checkActiveResourcePlugins(projectPath: string): Promise<boolean> {
+  try {
+    const projectSettings = await loadProjectSettings(projectPath);
+    const solutionSettings = projectSettings.solutionSettings;
+    if (projectSettings && solutionSettings && solutionSettings.activeResourcePlugins) {
+      return true;
+    } else {
+      return false;
+    }
+  } catch (error) {
+    return false;
```

Comment:
should we add some telemetry about why it fails?

Reply:
Maybe also add some telemetry to tell how many projects have invalid projectSettings.json